addons/website_sale/models/product_image.py

```diff
@@ -31,11 +34,56 @@ def _compute_can_image_1024_be_zoomed(self):
         for image in self:
             image.can_image_1024_be_zoomed = image.image_1920 and tools.is_image_size_above(image.image_1920, image.image_1024)
 
+    def set_thumbnail_image(self, video_url):
+        self.ensure_one()
+        status = False
+        if self.embed_code:
+            source = get_video_embed_code(self.video_url, source_with_document_id=True)
+            if source and hasattr(self, '_parse_%s_thumbnail' % source[0]):
+                status = getattr(self, '_parse_%s_thumbnail' % source[0])(source[1])
+            # In case of unknown provider or private video we set default image.
+            if not status:
+                #set a default image
+                image_path = get_module_resource('web', 'static/src/img', 'placeholder.png')
+                image = base64.b64encode(open(image_path, 'rb').read())
+                status = tools.image_process(image)
+        return status
+
+    @api.onchange('video_url')
+    def _onchange_video_url(self):
+        self.image_1920 = self.set_thumbnail_image(self.video_url)
+
     @api.depends('video_url')
     def _compute_embed_code(self):
         for image in self:
             image.embed_code = get_video_embed_code(image.video_url)
 
+    def _parse_youtube_thumbnail(self, document_id):
```

Comment:
With changes proposed above this will go to `get_video_thumbnail` tools's method in a switch case statement.